pylint/checkers/utils.py

```diff
@@ -1476,6 +1480,46 @@ def is_registered_in_singledispatch_function(node: nodes.FunctionDef) -> bool:
     return False
 
 
+def find_infered_fn_from_register(node: nodes.NodeNG) -> nodes.FunctionDef | None:
+    # func.register are function calls or register attributes
+    # when the function is annotated with types
+    if isinstance(node, nodes.Call):
+        func = node.func
+    elif isinstance(node, nodes.Attribute):
+        func = node
+    else:
+        return None
+
+    if not isinstance(func, nodes.Attribute) or func.attrname != "register":
+        return None
+
+    func_def = safe_infer(func.expr)
+    if not isinstance(func_def, nodes.FunctionDef):
+        return None
+
+    return func_def
+
+
+def is_registered_in_singledispatchmethod_function(node: nodes.FunctionDef) -> bool:
+    """Check if the given function node is a 'singledispatchmethod' function."""
+
+    singledispatchmethod_qnames = (
+        "functools.singledispatchmethod",
+        "singledispatch.singledispatchmethod",
+    )
+
+    if not isinstance(node, nodes.FunctionDef):
```

Comment:
This should be unreachable based on the typing, right?

Reply:
It should, I think I copied this line from the other function that does something similar. I'll remove.

Reply:
I mean, I would also be okay with widening the typing and having this accept all types of `nodes`. Although I wouldn't know why you would ever call this with anything other than a `FunctionDef`. Up to you~